### PR TITLE
docs: add custom readme content

### LINKED ISSUE #1086: Add agent usage information to the README.
The README should have information about the usage and configuration of the agent as well as what limitations exist.  The current README is autogenerated and lacks the required depth of content.

### PR DESCRIPTION
**Pull request by mctavish**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/cloud-debug-nodejs/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1086 🦕

### CODE REVIEW

File: .readme-partials.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+introduction: |-
+  > This module provides Cloud Debugger support for Node.js applications.
+  Cloud Debugger is a feature of Google Cloud Platform that lets you debug your
+  applications in production without stopping or pausing your application.
+
+body: |-
+  ## Debugger Agent Settings
+
+  You can customize the behaviour of the automatic debugger agent. See
```
Comment (fatoboist) [inline .readme-partials.yaml:9]: Generally avoid statements  that start with "You can..." re-word as "To customize the behavior of the automatic debugger agent see..."

File: .readme-partials.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  [the agent configuration][config-ts] for a list of possible configuration
+  options. These options can be passed in the `options` object passed to the
+  `start` function.
+
+  ```js
```
Comment (fatoboist) [inline .readme-partials.yaml:14]: It's not clear what this code snippet is showing. Add a descriptive sentence before the code sample stating it's purpose.

File: .readme-partials.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      service: 'my-service',
+      version: 'version-1'
+    },
+    capture: { maxFrames: 20, maxProperties: 100 }
+  });
+  ```
+
+  ## Using the Debugger
+
+  Once your application is running (deployed, or elsewhere), you should be able
```
Comment (fatoboist) [inline .readme-partials.yaml:30]: what do you mean by 'elsewhere'? I would either define or remove the content in ()

Comment (fatoboist) [reply .readme-partials.yaml:30]: remove "should be able" and rephrase as a more concrete "once your application is running, you can use the Debug UI in your Cloud Developer console.

File: .readme-partials.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  to use the [Debug UI][debug-tab] in your Cloud
+  [developer console][dev-console]. You can find the Debug UI in the
+  'Operations -> Debug' section in the navigation panel, or by simply searching
+  for 'Debug' in the cloud console.
+
+  If you have the source available, you can set a snapshot by clicking in the
```
Comment (fatoboist) [inline .readme-partials.yaml:36]: This would flow better as a numbered list of instructions I think. For example:

To take a snapshot with the debugger:

1. Click in the gutter (line number area) or enter a filename and line number in the snapshot panel
2. The debugger inserts a momentary breakpoint at the specified location in your code in the running instance of your application.

File: .readme-partials.yaml (hunk 1 above)
Comment (fatoboist) [inline .readme-partials.yaml:9]: Generally avoid statements that start with "You can..." re-word as "To customize the behavior of the automatic debugger agent see..."

File: .readme-partials.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  applications in production without stopping or pausing your application.
+
+body: |-
+  ## Debugger Agent Settings
+
+  You can customize the behaviour of the automatic debugger agent. See
+  [the agent configuration][config-ts] for a list of possible configuration
+  options. These options can be passed in the `options` object passed to the
+  `start` function.
+
```
Comment (fatoboist) [inline .readme-partials.yaml:13]: It's not clear what this code snippet is showing. Add a descriptive sentence before the code sample stating its purpose.  Something like "The following code sample does X"

File: .readme-partials.yaml (hunk 3 above)
Comment (fatoboist) [inline .readme-partials.yaml:30]: what do you mean by 'elsewhere'? I would either define or remove the content in ()

Comment (fatoboist) [inline .readme-partials.yaml:30]: remove "should be able" and rephrase as a more concrete "once your application is running, you can use the Debug UI in your Cloud Developer console.

File: .readme-partials.yaml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  ## Firebase Realtime Database backend
+
+  The Cloud Debugger API is deprecated and will be turned down in May 2023.
+
+  You can use Firebase Realtime Database for data persistence as an
+  alternative.
+
+  ### Enabling the agent
+
```
Comment (fatoboist) [inline .readme-partials.yaml:60]: Add a paragraph stating what the following code sample does. e.g. "To enable the agent, add the following to [file it should be added to]."

File: .readme-partials.yaml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  ```js
+  require('@google-cloud/debug-agent').start({
+    useFirebase: true,
+    // optional; will use https://my-project-id-cdbg.firebase.io.com if not provided.
```
Comment (fatoboist) [inline .readme-partials.yaml:64]: I would strongly recommend taking these comments out of the code sample and instead adding several explanations after the sample.
e.g.  something along the lines of 

"The following params are optional:

- `firebaseDbUrl` - `https://PROJECT_ID-cdbg.firebase.io.com` will be used if not provided. where PROJECT_ID is your project ID.
- firebaseKeyPath - Default google application credentials are used if not provided"

File: .readme-partials.yaml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    firebaseKeyPath: 'path/to/service_account.json',
+  });
+  ```
+
+  ### Using the Debugger
+
+  Using the Debugger with the Firebase Realtime Database backend requires using
+  the Snapshot Debugger CLI.
+
+  See [full documentation here][snapshot-debugger-readme].
```
Comment (fatoboist) [inline .readme-partials.yaml:76]: suggest changing to "See the full Snapshot Debugger CLI documentation."

File: .readme-partials.yaml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+  ## Limitations and Requirements
+
+  > Note: There is a known issue where enabling the agent may trigger memory
+  leaks.  See [#811](https://github.com/googleapis/cloud-debug-nodejs/issues/811)
+
+  * The root directory of your application needs to contain a `package.json`
+    file.
+  * You can set snapshot conditions and watch expressions to be evaluated in
```
Comment (fatoboist) [inline .readme-partials.yaml:85]: I would reword this to start with the Privacy issues. something like:

"Privacy issues can be created by setting snapshot conditions that watch expressions evaluated in the context of your application. You may be able to view sensitive user data when viewing the values of variables."

**Review by fatoboist — COMMENTED** [review]
Left some suggestions. Definitely need to state what each code sample is doing even if you think it's obvious from the heading before it :)

File: .readme-partials.yaml (hunk 4 above)
Comment (mctavish) [reply .readme-partials.yaml:36]: Done, and yes, it does read easier now.

File: .readme-partials.yaml (hunk 1 above)
Comment (mctavish) [reply .readme-partials.yaml:9]: Unfortunate that this documentation is littered with "you can", eh?  I've reworked most of them now.

File: .readme-partials.yaml (hunk 5 above)
Comment (mctavish) [reply .readme-partials.yaml:13]: Added "The following code sample shows how to set some custom options when starting the debugger agent."

File: .readme-partials.yaml (hunk 3 above)
Comment (mctavish) [reply .readme-partials.yaml:30]: Removed.

Comment (mctavish) [reply .readme-partials.yaml:30]: Done

File: .readme-partials.yaml (hunk 6 above)
Comment (mctavish) [reply .readme-partials.yaml:60]: Done.

File: .readme-partials.yaml (hunk 7 above)
Comment (mctavish) [reply .readme-partials.yaml:64]: Good suggestion.  I like it.

File: .readme-partials.yaml (hunk 8 above)
Comment (mctavish) [reply .readme-partials.yaml:76]: Done

File: .readme-partials.yaml (hunk 9 above)
Comment (mctavish) [reply .readme-partials.yaml:85]: Done

**Review by jasonborg — APPROVED** [review]
